### ISSUE TITLE
HexChat crashes on windows when using Chinese characters in timestamps.

### ISSUE THREAD

**Issue by lachlankidson** [hexchat/hexchat#1809]
Produced on HexChat 2.12.1 x64 on Windows 10.

HexChat crashes upon clicking ok in Appearance options after using Chinese characters in the 'time stamp format' field.

i.e., %Y年%m月%d日 which should produce 2016年08月30日

Also reproducible for languages such as Hebrew, suspect issue with strftime implementation.

**Comment by TingPing** [comment]
Yes `strftime()` is locale dependent. We have a `strftime_utf8()` that makes use of `g_date_strftime()` but this cannot be used everywhere because it doesn't support all formatting that you need for chat timestamps.
